doc/source/user_guide/tutorials/mesh/create_a_mesh_from_scratch.rst

```diff
@@ -0,0 +1,158 @@
+.. _ref_tutorials_create_a_mesh_from_scratch:
+
+==========================
+Create a mesh from scratch
+==========================
+
+.. include:: ../../../links_and_refs.rst
+
+This tutorial demonstrates how to build a |MeshedRegion| from the scratch.
+
+The mesh object in DPF is a |MeshedRegion|. You can create your own |MeshedRegion| object and use it
+with DPF operators. The ability to use scripting to create any DPF entity means
+that you are not dependent on result files and can connect the DPF environment
+with any Python tool.
+
+In this tutorial, we create a parallel piped mesh made of linear hexa elements.
+
+:jupyter-download-script:`Download tutorial as Python script<create_a_mesh_from_scratch>`
```

Comment:
Just to make sure, this role directive is provided by the Jupyter Sphinx extension, right? Does it combine all the code-blocks into a big notebook file?

Reply:
Yes, exactly

Reply:
Thanks for explaining. Then, it makes sense that we keep those comments within the code-blocks as discussed.